### PR TITLE
Minor, refactor

### PR DESCRIPTION
**Pull request by RupengWang**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

- disbale setting kylin config
- add check for database name
- add check for command parameter

## Types of changes

What types of changes does your code introduce to Kylin?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have create an issue on [Kylin's jira](https://issues.apache.org/jira/browse/KYLIN), and have described the bug/feature there in detail
- [ ] Commit messages in my PR start with the related jira ID, like "KYLIN-0000 Make Kylin project open-source"
- [ ] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If this change need a document change, I will prepare another pr against the `document` branch
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at user@kylin or dev@kylin by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: core-common/src/main/java/org/apache/kylin/common/util/CliCommandExecutor.java [hunk 1]
```diff
@@ -195,10 +197,14 @@ public static String checkParameterWhiteList(String commandParameter) {
         return checkParameter(commandParameter, COMMAND_WHITE_LIST);
     }
 
+    public static String checkHivePropertyWhiteList(String hiveProperty) {
```
Comment (hit-lacus) [inline core-common/src/main/java/org/apache/kylin/common/util/CliCommandExecutor.java:200]: White or black, it is a question!😁

File: server-base/src/main/java/org/apache/kylin/rest/controller/AdminController.java [hunk 2]
```diff
@@ -118,11 +116,6 @@ public void cleanupStorage() {
         adminService.cleanupStorage();
     }
 
-    @RequestMapping(value = "/config", method = { RequestMethod.PUT }, produces = { "application/json" })
```
Comment (shaofengshi) [inline server-base/src/main/java/org/apache/kylin/rest/controller/AdminController.java:121]: I'm not quite sure how many users are using this API. If we can provide another way to enable it (on their own risk), that can be more flexible.

Comment (RupengWang) [reply server-base/src/main/java/org/apache/kylin/rest/controller/AdminController.java:121]: Yes I aggree, it's better to enable or disable this API with kylin.properties configuration. I'm working on it.

Comment (RupengWang) [reply server-base/src/main/java/org/apache/kylin/rest/controller/AdminController.java:121]: Currently the function that setting config from the front end only can be enable or disable at the backend by modify the property  "kylin.web.set-config-enable" at kylin.properties and it's disable by default. 
If "kylin.web.set-config-enable" is set to false, kylin will throw an exception when user try to  set config from website or restful API.
image

**Pull request merged**